classes/components/forms/context/PKPEmailSetupForm.inc.php

```diff
@@ -37,33 +37,38 @@ public function __construct($action, $locales, $context) {
 		$this->action = $action;
 		$this->locales = $locales;
 
-		$canEnvelopeSender = \Config::getVar('email', 'allow_envelope_sender');
-
 		$this->addField(new FieldRichTextarea('emailSignature', [
-				'label' => __('manager.setup.emailSignature'),
-				'tooltip' => __('manager.setup.emailSignature.description'),
-				'value' => $context->getData('emailSignature'),
-				'preparedContent' => [
-					'contextName' => $context->getLocalizedName(),
-					'senderName' => __('email.senderName'),
-					'senderEmail' => __('email.senderEmail'),
-					'mailingAddress' => htmlspecialchars(nl2br($context->getData('mailingAddress'))),
-					'contactEmail' => htmlspecialchars($context->getData('contactEmail')),
-					'contactName' => htmlspecialchars($context->getData('contactName')),
-				]
-			]));
+			'label' => __('manager.setup.emailSignature'),
+			'tooltip' => __('manager.setup.emailSignature.description'),
+			'value' => $context->getData('emailSignature'),
+			'preparedContent' => [
+				'contextName' => $context->getLocalizedName(),
+				'senderName' => __('email.senderName'),
+				'senderEmail' => __('email.senderEmail'),
+				'mailingAddress' => htmlspecialchars(nl2br($context->getData('mailingAddress'))),
+				'contactEmail' => htmlspecialchars($context->getData('contactEmail')),
+				'contactName' => htmlspecialchars($context->getData('contactName')),
+			]
+		]));
+
+		$this->buildEnveloperSenderField($context);
+	}
+
+	protected function buildEnveloperSenderField($context) {
+		$canEnvelopeSender = \Config::getVar('email', 'allow_envelope_sender');
 
 		if ($canEnvelopeSender) {
 			$this->addField(new FieldText('envelopeSender', [
 				'label' => __('manager.setup.emailBounceAddress'),
 				'tooltip' => __('manager.setup.emailBounceAddress.description'),
 				'value' => $context->getData('envelopeSender'),
 			]));
-		} else {
-			$this->addField(new FieldHTML('envelopeSender', [
-				'label' => __('manager.setup.emailBounceAddress'),
-				'description' => __('manager.setup.emailBounceAddress.disabled'),
-			]));
+			return;
 		}
+
+		$this->addField(new FieldHTML('envelopeSender', [
+			'label' => __('manager.setup.emailBounceAddress'),
+			'description' => __('manager.setup.emailBounceAddress.disabled'),
+		]));
```

Comment:
It doesn't look like there have been any changes to this file, so can you revert it to what it was before?

Reply:
Followed the Uncle Bob's ["Boy Scout Rule"](https://biratkirat.medium.com/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385). ;)

Reply:
Ok, you can leave it in. But beware of premature optimization. I'm wary of code churn like this when there isn't a significant benefit. There's always a risk of introducing bugs, however small, and I've caused havoc with little improvements like this before. :joy: